lib/pi_piper/pin.rb

```diff
@@ -137,53 +132,23 @@ def changed?
     # Blocks until a logic level change occurs. The initializer option 
     # `:trigger` modifies what edge this method will release on.
     def wait_for_change
-      fd = File.open(value_file, "r")
-      File.open(edge_file, "w") { |f| f.write("both") }
-      loop do
-        fd.read
-        IO.select(nil, nil, [fd], nil)
-        read
-        if changed?
-          next if @trigger == :rising and value == 0
-          next if @trigger == :falling and value == 1
-          break
-        end
-      end
+      Platform.driver.pin_wait_for(@pin, @trigger)
     end
-
+    
     # Reads the current value from the pin. Without calling this method 
     # first, `value`, `last_value` and `changed?` will not be updated.
     # 
     # In short, you must call this method if you are curious about the 
     # current state of the pin.
     def read
-      fail PiPiper::PinError, "Pin #{@pin} already released" if released?
```

Comment:
I think we still need this here, what happens if someone releases the pin and then calls `read`. The `@last_value` could potentially overwrite the last actual value.